plugins/star-rating/frontend/public/javascripts/countly.views.js

```diff
@@ -611,360 +619,492 @@ window.starView = countlyView.extend({
+                            return jQuery.i18n.map['feedback.bottom-left'];
+                            break;
+                        case 'bright':
+                            return jQuery.i18n.map['feedback.bottom-left'];
+                            break;
+                        default:
+                            return jQuery.i18n.map['feedback.middle-right'];
+                    }
                 },
-                {
-                    "mData":function (row) {
-                        var target_pages = "";
-                        eval(row.target_pages).forEach(function(page) { target_pages += "\n" + page;})
-                        return target_pages.trim();
-                    },
-                    sType: "string",
-                    "sTitle": jQuery.i18n.map["feedback.target-pages"]
+                sType: "string",
+                "sTitle": jQuery.i18n.map["common.location"]
+            }, {
+                "mData": function(row) {
+                    var target_pages = "";
+                    if (typeof row.target_pages == "string") row.target_pages = eval(row.target_pages);
+                    row.target_pages.forEach(function(page) {
+                        target_pages += "\n" + page;
+                    })
+                    return target_pages.trim();
                 },
-                {
-                    "mData":function (row) {
-                        var td = JSON.parse(row.target_devices);
-                        if (td.length == 2) {
-                            return td[0].substr(0,1).toUpperCase()+td[0].substr(1,td[0].length-1) + " and " + td[1];
-                        } else if(td.length == 3) {
-                            return "Mobile" + ", " + td[1] + " and " + td[2];
-                        } else if (td.length == 1){
-                            return td[0].replace(/\b\w/g, function(l) { l.toUpperCase()});
-                        } else return "No device selected.";
-                    },
-                    sType: "string",
-                    "sTitle": jQuery.i18n.map["feedback.target-devices"],
-                    "sWidth":"20%"
-                }
-            ];
+                sType: "string",
+                "sTitle": jQuery.i18n.map["feedback.target-pages"]
+            }, {
+                "mData": function(row) {
+                    if (typeof row.target_devices == "string") var td = JSON.parse(row.target_devices);
+                    else var td = row.target_devices;
+                    
+                    var keys = Object.keys(td);
+                    var vals = Object.values(td);
+                    var atLeastOneDevice = false;
+                    var deviceText = '';
+
+                    keys.forEach(function(key) {
+                        if(td[key]) {
+                            deviceText += key + ' ';  
+                            atLeastOneDevice = true;
+                        } 
+                    })
```

Comment:
This can be replace by this (without using unsupported functions as `Object.keys` and `Object.values`)

```
for(var key in td){
    if(td[key]) {
        deviceText += key + ' ';  
        atLeastOneDevice = true;
    } 
}
```